prow/sidecar/run.go

```diff
@@ -138,15 +172,26 @@ func (o Options) doUpload(spec *downwardapi.JobSpec, passed, aborted bool) error
 		result = "FAILURE"
 	}
 
+	// TODO(krzyzacy): Unify with downstream spyglass definition
 	finished := struct {
-		Timestamp int64  `json:"timestamp"`
-		Passed    bool   `json:"passed"`
-		Result    string `json:"result"`
+		Timestamp int64                  `json:"timestamp"`
+		Passed    bool                   `json:"passed"`
+		Result    string                 `json:"result"`
+		Metadata  map[string]interface{} `json:"metadata,omit_empty"`
```

Comment:
Technically I think this should be https://golang.org/pkg/encoding/json/#RawMessage but we can do in a follow-up